echo/wavefront-tracing-utils.go

```diff
@@ -0,0 +1,173 @@
+		if err != nil {
+			return err
+		}
+	} else {
+		// Create the proxy sender
+		sender, err = getWavefrontProxySender(server, flushIntervalSeconds)
+		if err != nil {
+			return err
+		}
+	}
+	appTags := wfApplication.New(tracerConfig.Application, tracerConfig.Service)
+	if tracerConfig != nil {
+		if len(tracerConfig.Cluster) > 0 {
+			appTags.Cluster = tracerConfig.Cluster
+		}
+		if len(tracerConfig.Shard) > 0 {
+			appTags.Shard = tracerConfig.Shard
+		}
+		if tracerConfig.CustomApplicationTags.StaticTags != nil && len(tracerConfig.CustomApplicationTags.StaticTags) > 0 {
+			for k, v := range tracerConfig.CustomApplicationTags.StaticTags {
+				appTags.CustomTags[k] = v
+			}
+		}
+	}
+	wfReporter := reporter.New(sender, appTags, reporter.Source(tracerConfig.Source))
+	clReporter := reporter.NewConsoleSpanReporter(tracerConfig.Source)
+	reporter := reporter.NewCompositeSpanReporter(wfReporter, clReporter)
+	var durationSampler tracer.DurationSampler
+	var rateSampler tracer.RateSampler
+
+	//Initialising tracer with RateSampler and DurationSampler if given in conifg
+	//If values for RateSampler and DurationSampler is not given, it initalises standard tracer with default values
+	if tracerConfig.DurationSampler > 0 && tracerConfig.RateSampler > 0 {
+		durationSampler = tracer.DurationSampler{Duration: time.Duration(tracerConfig.DurationSampler) * time.Second}
+		rateSampler = tracer.RateSampler{Rate: tracerConfig.RateSampler}
+
+		Tracer = wfTracer.New(reporter, wfTracer.WithSampler(rateSampler), wfTracer.WithSampler(durationSampler))
+	} else if tracerConfig.RateSampler > 0 {
+		rateSampler = tracer.RateSampler{Rate: tracerConfig.RateSampler}
+
+		Tracer = wfTracer.New(reporter, wfTracer.WithSampler(rateSampler))
+	} else if tracerConfig.DurationSampler > 0 {
+		durationSampler = tracer.DurationSampler{Duration: time.Duration(tracerConfig.DurationSampler) * time.Second}
+
+		Tracer = wfTracer.New(reporter, wfTracer.WithSampler(durationSampler))
+	} else {
+		Tracer = wfTracer.New(reporter)
+	}
+
+	//Instantiating Global Tracer
+	opentracing.InitGlobalTracer(Tracer)
+
+	//Enabling Middleware
+	echoWeb.Use(middlewareTracing)
+	log.Println("Tracer Initialized...")
+	return nil
+}
+
+//StartTraceSpan looks for existing context from the headers of the request.
+//If context is found, it starts a child span to the parent span,
+//Else it starts a root span.
+//Tags are added to the respective spans created.
+//If parent span exists parent span id is returned along with child span otherwise root span and "" is returned
+func StartTraceSpan(c echo.Context, appSpecificOperationName string, tags map[string]string) (opentracing.Span, string, error) {
+	var serverSpan opentracing.Span
+	var parentSpanID string
+	wireContext, err := opentracing.GlobalTracer().Extract(
+		opentracing.HTTPHeaders,
+		opentracing.HTTPHeadersCarrier(c.Request().Header))
+	if err != nil {
+		log.Println(err)
```

Comment:
Should this not return the error?

Reply:
No, this err is for notifying the user that there is no current context in the Call. And notifying the user that we are starting a new root span. This error will not break the flow.